neo/SmartContract/NativeContracts/NeoToken.cs

```diff
@@ -0,0 +1,406 @@
+﻿using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using VMArray = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract
+{
+    partial class NeoService
+    {
+        private static readonly string[] NeoToken_SupportedStandards = { "NEP-5", "NEP-10" };
+        private const string NeoToken_Name = "NEO";
+        private const string NeoToken_Symbol = "neo";
+        private const int NeoToken_Decimals = 0;
+        private static readonly BigInteger NeoToken_DecimalsFactor = BigInteger.Pow(10, NeoToken_Decimals);
+        private static readonly BigInteger NeoToken_TotalAmount = 100000000 * NeoToken_DecimalsFactor;
+
+        private const byte NeoToken_Prefix_Initialized = 11;
+        private const byte NeoToken_Prefix_Account = 20;
+        private const byte NeoToken_Prefix_Validator = 33;
+        private const byte NeoToken_Prefix_ValidatorsCount = 15;
+
+        private bool NeoToken_Main(ExecutionEngine engine)
+        {
+            if (!new UInt160(engine.CurrentContext.ScriptHash).Equals(Blockchain.NeoToken.ScriptHash))
+                return false;
+            string operation = engine.CurrentContext.EvaluationStack.Pop().GetString();
+            VMArray args = (VMArray)engine.CurrentContext.EvaluationStack.Pop();
+            StackItem result;
+            switch (operation)
+            {
+                case "supportedStandards":
+                    result = NeoToken_SupportedStandards.Select(p => (StackItem)p).ToList();
+                    break;
+                case "name":
+                    result = NeoToken_Name;
+                    break;
+                case "symbol":
+                    result = NeoToken_Symbol;
+                    break;
+                case "decimals":
+                    result = NeoToken_Decimals;
+                    break;
+                case "totalSupply":
+                    result = NeoToken_TotalAmount;
+                    break;
+                case "balanceOf":
+                    result = NeoToken_BalanceOf(engine, args[0].GetByteArray());
+                    break;
+                case "transfer":
+                    result = NeoToken_Transfer(engine, args[0].GetByteArray(), args[1].GetByteArray(), args[2].GetBigInteger());
+                    break;
+                case "initialize":
+                    result = NeoToken_Initialize(engine);
+                    break;
+                case "unclaimedGas":
+                    result = NeoToken_UnclaimedGas(args[0].GetByteArray(), (uint)args[1].GetBigInteger());
+                    break;
+                case "registerValidator":
+                    result = NeoToken_RegisterValidator(args[0].GetByteArray());
```

Comment:
Isn’t it supposed to cost something to register a validator?

Reply:
> Note: The price will be set in separate PR.